scripts/verify-pkg-main.js

```diff
@@ -36,31 +36,73 @@ if (!isValidCwd) {
 }
 
 let invalidMains = 0;
+let invalidModules = 0;
+let invalidTypes = 0;
 globSync('packages/*/package.json').forEach((jsonPath) => {
-  const packageInfo = JSON.parse(fs.readFileSync(jsonPath));
-  if (!packageInfo.main) {
+  const packageJson = JSON.parse(fs.readFileSync(jsonPath));
+  if (!packageJson.main) {
     return;
   }
+  verifyPath(packageJson, jsonPath, 'main');
+  verifyPath(packageJson, jsonPath, 'module');
+  verifyPath(packageJson, jsonPath, 'types');
+});
+
+if (invalidMains > 0 || invalidModules > 0 || invalidTypes > 0) {
+  if (invalidMains > 0) {
+    logErrorAndExit(`${invalidMains} incorrect main property values found; please fix.`);
+  }
+  if (invalidModules > 0) {
+    logErrorAndExit(`${invalidModules} incorrect module property values found; please fix.`);
+  }
+  if (invalidTypes > 0) {
+    logErrorAndExit(`${invalidTypes} incorrect types property values found; please fix.`);
+  }
+} else {
+  console.log('Success: All packages with main, module, types properties reference appropriate files!');
+}
 
-  const mainPath = path.join(path.dirname(jsonPath), packageInfo.main);
+/**
```

Comment:
Nit: Move these functions up above the immediately-executing code that uses them